src/pages/index.tsx

```diff
@@ -1,18 +1,33 @@
+import { useEffect, useState } from 'react'
 import Link from 'next/link'
 
 import Header from '@/components/Header'
 import Summary from '@/components/Home/summary'
 import NavBar from '@/components/NavBar'
+import { withProtected } from '@/components/PrivateRoute'
 import TopNav from '@/components/TopNav'
 import Button from '@/components/UI/button'
 import useUser from '@/hooks/user'
 
 const Home = () => {
-  const { isSuccess, data: currentUser } = useUser()
+  const { data: currentUser } = useUser()
 
-  const welcomeMessage = isSuccess
-    ? `Welcome, ${currentUser?.name}!`
-    : 'Welcome!'
+  const [welcomeMessage, setWelcomeMessage] = useState('Welcome!')
```

Comment:
You don't need to call `useState` for any of this, as the `useUser` hook takes care of state for you (via React Query). If you revert this to the method used from my branch you can see what I mean.